src/components/tree-view/TreeView.tsx

```diff
@@ -1,48 +1,30 @@
 import { NodeApi, NodeRendererProps, Tree } from "react-arborist";
 import { TreeViewSeparator } from "./TreeViewSeparator";
+import useResizeObserver from "use-resize-observer";
+
 import {
   BaseTreeViewData,
   TreeViewDataType,
   TreeViewMoveModeEnum,
   TreeViewMoveResult,
+  TreeViewNodeTypeEnum,
 } from "./types";
-
-export const treeDemoData = [
-  { id: "1", name: "Unread" },
-  { id: "2", name: "Threads" },
-  { id: "2.1", name: "load children", childrenCount: 1, children: [] },
-  {
-    id: "3",
-    name: "Chat Rooms",
-    children: [
-      { id: "c1", name: "General" },
-      { id: "c2", name: "Random" },
-      { id: "c3", name: "Open Source Projects" },
-    ],
-  },
-  {
-    id: "4",
-    name: "Direct Messages",
-    children: [
-      { id: "d1", name: "Alice" },
-      { id: "d2", name: "Bob" },
-      { id: "d3", name: "Charlie" },
-    ],
-  },
-];
+import { isNode } from "./utils";
 
 export type TreeViewProps<T> = {
   treeData: TreeViewDataType<T>[];
-  width?: number | string;
   selectedNodeId?: string;
   rootNodeId: string;
+  canDrop?: (args: {
+    parentNode: NodeApi<BaseTreeViewData<T>> | null;
+    dragNodes: NodeApi<BaseTreeViewData<T>>[];
+    index: number;
+  }) => boolean;
+  canDrag?: (node: BaseTreeViewData<T>) => boolean;
+  handleMove?: (result: TreeViewMoveResult) => void;
```

Comment:
You might call me nit-picker-from-hell but to stay aligned with react philisophy what do your think about using "onMove"? 😬